src/ImageSharp/Formats/Gif/GifDecoderCore.cs

```diff
@@ -158,6 +157,11 @@ public Image<TPixel> Decode<TPixel>(BufferedReadStream stream, CancellationToken
             return image;
         }
 
+        /// <inheritdoc />
+        public void Decode<TPixel>(BufferedReadStream stream, Image<TPixel> image, CancellationToken cancellationToken)
+            where TPixel : unmanaged, IPixel<TPixel>
+            => throw new NotSupportedException("The current decoder doesn't support decoding to a target image");
```

Comment:
Why not decode a single frame?

Reply:
What to do in the case of a multi-frame image was something I was unsure about in the initial discussion proposal as well. Decoding the first frame is definitely what I'd expect when using the API. It might also make sense to offer additional overloads for specifying a frame index, but I'm not completely sold on that being the best approach.

Reply:
I would probably leave it at frame 0 for now. Having the value settable might well complicate things to much for a single PR.